src/runtime/Types/ClassBase.cs

```diff
@@ -529,6 +529,27 @@ static NewReference tp_call_impl(BorrowedReference ob, BorrowedReference args, B
             return callBinder.Invoke(ob, args, kw);
         }
 
+        static NewReference DoConvert(BorrowedReference ob)
+        {
+            var self = (CLRObject)GetManagedObject(ob)!;
+            using var python = self.inst.ToPython();
+            return python.NewReferenceOrNull();
```

Comment:
Avoid creating `PyObject` by directly calling `Runtime.PyInt_FromInt64` and/or `Runtime.PyLong_FromUnsignedLongLong`

Reply:
The code is much simpler this way, we can optimise it later if it's necessary.

Reply:
The problem with this approach is that it will generate .NET exceptions instead of Python errors (which I don't expect to happen here though), so Python will have problems unwinding the stack.

Reply:
Okay, got it. The `Converter.To<something>` functions would throw as well (crashing the process), but I think it's guaranteed in the way that they are called that this can't happen.